did/pkh.go

```diff
@@ -82,14 +91,58 @@ func (d DIDPKH) Method() Method {
 	return PKHMethod
 }
 
-// GetNetwork returns the network by finding the network prefix in the did
-func GetNetwork(did DIDPKH) (*Network, error) {
-	for network, prefix := range didPKHNetworkPrefixMap {
-		if strings.Contains(string(did), prefix[0]+":") {
-			return &network, nil
+// GetDIDPKHPrefixForNetwork returns the did:pkh prefix for a given network
+func GetDIDPKHPrefixForNetwork(n Network) (string, error) {
+	switch n {
+	case Bitcoin:
+		return BitcoinNetworkPrefix, nil
+	case Ethereum:
+		return EthereumNetworkPrefix, nil
+	case Polygon:
+		return PolygonNetworkPrefix, nil
+	}
+	return "", fmt.Errorf("unsupported did:pkh network: %s", n)
+}
+
+// GetDIDPKHNetworkForPrefix returns the did:pkh network for a given prefix
+func GetDIDPKHNetworkForPrefix(p string) (Network, error) {
+	switch p {
+	case BitcoinNetworkPrefix:
+		return Bitcoin, nil
+	case EthereumNetworkPrefix:
+		return Ethereum, nil
+	case PolygonNetworkPrefix:
+		return Polygon, nil
+	}
+	return "", fmt.Errorf("unsupported did:pkh prefix: %s", p)
+}
+
+// GetDIDPKHNetworkForDID returns the network for a given did:pkh
+func GetDIDPKHNetworkForDID(did string) (Network, error) {
+	prefixes := GetDIDPKHNetworkPrefixes()
+	for _, prefix := range prefixes {
+		if strings.Contains(did, prefix+":") {
+			return GetDIDPKHNetworkForPrefix(prefix)
 		}
 	}
-	return nil, util.LoggingNewError("network not supported")
+	return "", fmt.Errorf("could not find network for did:pkh DID: %s", did)
+}
+
+// GetVerificationTypeForNetwork returns the verification key type for a given network
+func GetVerificationTypeForNetwork(n Network) (string, error) {
```

Comment:
Nice, we don't have to have a mapping anymore, there could be a case where we add a new pkh but forget to add it here in the GetVerificationTypeForNetwork function, thats why I opt to have one source of truth mapping. But this works good too

Reply:
good point, let me see how to define this once